test/src/integration/onChain.test.ts

```diff
@@ -0,0 +1,215 @@
+// Copyright 2018 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import { TestHelper } from "codechain-test-helper/lib/testHelper";
+import CodeChain from "../helper/spawn";
+import { H256 } from "codechain-sdk/lib/core/H256";
+import { H160 } from "codechain-sdk/lib/core/H160";
+import { U256 } from "codechain-sdk/lib/core/U256";
+import { Header } from "codechain-test-helper/lib/cHeader";
+
+describe("Test onChain communication", () => {
+    let nodeA: CodeChain;
+
+    beforeEach(async () => {
+        nodeA = new CodeChain({ logFlag: true });
+        await nodeA.start();
+    });
+
+    afterEach(async () => {
+        await nodeA.clean();
+    });
+
+    test("OnChain parcel propagation test", async () => {
+        const TH = new TestHelper("0.0.0.0", nodeA.port);
+        await TH.establish();
+
+        const sdk = nodeA.sdk;
+
+        const ACCOUNT_SECRET =
+            process.env.ACCOUNT_SECRET ||
+            "ede1d4ccb4ec9a8bbbae9a13db3f4a7b56ea04189be86ac3a6a439d9a0a1addd";
+        const parcel = sdk.core.createPaymentParcel({
+            recipient: "tccqruq09sfgax77nj4gukjcuq69uzeyv0jcs7vzngg",
+            amount: 10000
+        });
+        const signedparcel = parcel.sign({
+            secret: ACCOUNT_SECRET,
+            fee: 10,
+            nonce: 0
+        });
+
+        await TH.sendEncodedParcel([signedparcel.toEncodeObject()]);
+        const block = await sdk.rpc.chain.getBlock(1);
+        expect(block).toEqual(expect.anything());
+
+        await TH.end();
+    });
+
+    test("OnChain block propagation test", async () => {
+        const TH = new TestHelper("0.0.0.0", nodeA.port);
+        await TH.establish();
+
+        const sdk = nodeA.sdk;
+
+        // Genesis block
```

Comment:
Could you replace it with a function?

We don't need to read the whole genesis block data at this point.

Reply:
Also for the block1 and block2.

Reply:
I made it as function